pkg/provider/resource_aws_connection.go

```diff
@@ -0,0 +1,113 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/api"
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/models"
+	"github.com/EnterpriseDB/terraform-provider-biganimal/pkg/utils"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"time"
+)
+
+type AWSConnectionResource struct{}
+
+func NewAWSConnectionResource() *AWSConnectionResource {
+	return &AWSConnectionResource{}
+}
+
+func (a *AWSConnectionResource) Schema() *schema.Resource {
+	return &schema.Resource{
+		Description: `The aws_connection resource is used to make connection between your project and AWS.
+o obtain the necessary input parameters, please refer to [Connect CSP](https://www.enterprisedb.com/docs/biganimal/latest/getting_started/02_connecting_to_your_cloud/connecting_aws/).`,
+
+		CreateContext: a.Create,
+		ReadContext:   a.Read,
+		UpdateContext: a.Update,
+		DeleteContext: a.Delete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Description: "Project ID of the project.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"external_id": {
+				Description: "Project Name of the project.",
```

Comment:
Please update the description of these fields. 
Also, it would be helpful to write how the external_id and role_arn should look like. You can check the descriptions of the `biganimal_cluster` resource for reference.

Reply:
Thanks! My negligence